porcupine/plugins/filetypes.py

```diff
@@ -252,6 +253,33 @@ def parse_filetype_name(name: str) -> FileType:
     )
 
 
+filetypes_var: tkinter.StringVar
+
+
+def _sync_filetypes_menu(event: object) -> None:
+    tab = get_tab_manager().select()
+
+    if isinstance(tab, tabs.FileTab):
+        try:
+            filetype_name = tab.settings.get("filetype_name", str)
+        except KeyError:
+            filetype_name = get_filetype_for_tab(tab).get("filetype_name")
```

Comment:
We want to keep the variable set to whatever is in the `filetype_name` setting of the current tab, or empty. To me, an obvious solution would be to sync when:

- Different tab is selected
- `filetype_name` setting of any `FileTab` changes (currently selected tab would be enough, but it's easier to bind on all `FileTab`s)

Now it syncs only when the tab changes, and if `filetype_name` hasn't been set yet, it predicts what it will be set to in the future. It probably works, but it feels unnecessarily fragile and weird to me.

Reply:
Ok, yeah I think I'm getting lost in what events to use. 
Tab selection is this event: "<<NotebookTabChanged>>?

And the filetype name change is the one you mentioned earlier: 
One possible solution is to run the sync whenever the filetype_name setting changes. We can do it by binding <<TabSettingChanged:filetype_name>> inside on_new_filetab().

Reply:
Yeah, the events can be confusing :)

- `<<NotebookTabChanged>>`: An event available on any `ttk.Notebook`. Runs when the current tab changes. Settings are not populated yet, because they are added in a callback that runs after a new tab is added.
- `<<TabSettingChanged:filetype_name>>`: Available on any `FileTab` (not the `TabManager`, which is a `ttk.Notebook`). Runs when the setting changes. This includes user selecting it in the menu (although that automatically updates the `StringVar` anyway), when the settings are set for the first time for a new or newly opened file, and in any other situation where the setting might change (e.g. in a future version of Porcupine).
- `get_tab_manager().add_filetab_callback()`: Not an event, but runs on newly added tabs. It currently sets their filetypes, but it could also bind so that the sync runs when the filetype changes.

Reply:
Ok so for my sync listener, should I bind to both <<NotebookTabChanged>> and <<TabSettingChanged:filetype_name>> and then take out the attempt to call get_filetype_for_tab?

Reply:
That would make the most sense to me :) Of course, you can do it in any way that works if you manage to convince me that your way makes sense or is better in some way.

Reply:
Ok, I could not get the <<TabSettingChanged:filetype_name>> to ever fire.
but I added sync call to on_new_filetab in the filetypes plugin and that seems to work
This is probably as close as I can get it so feel free to modify as needed